File: 2026/04-April/28-Tuesday/solution.go
import "sort"

func minOperations(grid [][]int, x int) int {
    nums := []int{}
 
    for _, row := range grid {
        nums = append(nums, row...)
    }

    sort.Ints(nums)

    base := nums[0] % x
    for _, num := range nums {
        if num%x != base {
            return -1
        }
    }

    median := nums[len(nums)/2]
    ops := 0

    for _, num := range nums {
        if num > median {
            ops += (num - median) / x
        } else {
            ops += (median - num) / x
        }
    }

    return ops
}

File: 2026/04-April/28-Tuesday/solution.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int minOperations(vector<vector<int>>& grid, int x) {
        vector<int> nums;

        for (auto& row : grid) {
            for (int num : row) {
                nums.push_back(num);
            }
        }

        sort(nums.begin(), nums.end());

        int base = nums[0] % x;
        for (int num : nums) {
            if (num % x != base) return -1;
        }

        int median = nums[nums.size() / 2];
        int ops = 0;

        for (int num : nums) {
            ops += abs(num - median) / x;
        }

        return ops;
    }
};
